lib/dry/configurable/null_config.rb

```diff
@@ -0,0 +1,40 @@
+module Dry
+  module Configurable
+
+    # @private
+    class NullConfig
+      def initialize
+        @schema = {}
+      end
+
+      def to_config
+        schema.each_with_object({}) do |(key, value), hash|
+          case value
+          when NullConfig
+            hash[key] = value.to_config
+          else
+            hash[key] = value
+          end
+        end
+      end
+
+      EQUAL_END = /=$/
+
+      def method_missing(method_name, *args, &block)
+        to_return = self
+        if method_name.to_s.match(EQUAL_END)
```

Comment:
Using `rindex` + `string[0, index]` might be faster:

### Ref

https://github.com/neopoly/nested_config/blob/7b8c1f7ccf9ad14ec02cb9fdefbf7fa86a925d14/lib/nested_config.rb#L79,L86

### Bench

```ruby
require "benchmark/ips"

EQUAL_END = /=$/

def mm(method_name)
  key = method_name.to_s

  if method_name.to_s.match(EQUAL_END)
    key = method_name.to_s.gsub(EQUAL_END, '').to_sym
  else
    # noop
  end
end

EQUAL = "=".freeze

def mm2(method_name)
  key = method_name.to_s
  index = key.rindex(EQUAL)
  if index
    key = key[0, index].to_sym
  else
    # noop
  end
end

p mm(:method)
p mm(:method=)

p mm2(:method)
p mm2(:method=)

Benchmark.ips do |x|
  x.report "mm getter" do
    mm(:method)
  end

  x.report "mm setter" do
    mm(:method=)
  end

  x.report "mm2 getter" do
    mm2(:method)
  end

  x.report "mm2 setter" do
    mm2(:method=)
  end

  x.compare!
end
```

### Results

```
nil
:method
nil
:method
Warming up --------------------------------------
           mm getter    83.252k i/100ms
           mm setter    21.563k i/100ms
          mm2 getter   140.865k i/100ms
          mm2 setter    96.049k i/100ms
Calculating -------------------------------------
           mm getter      1.455M (± 2.0%) i/s -      7.326M in   5.037411s
           mm setter    267.059k (± 2.5%) i/s -      1.337M in   5.009369s
          mm2 getter      2.860M (± 3.5%) i/s -     14.368M in   5.029904s
          mm2 setter      1.551M (± 3.2%) i/s -      7.780M in   5.020025s

Comparison:
          mm2 getter:  2860183.6 i/s
          mm2 setter:  1551423.7 i/s - 1.84x  slower
           mm getter:  1454968.8 i/s - 1.97x  slower
           mm setter:   267059.2 i/s - 10.71x  slower
```

Reply:
I believe `end_with?('=')` + `.chop` will be even faster :)

Reply:
@flash-gordon I like your approach :+1: which not also looks nicer but is tiny bit faster as well :rocket: 

## Bench

```ruby
require "benchmark/ips"

EQUAL = "=".freeze

def mm2(method_name)
  key = method_name.to_s
  index = key.rindex(EQUAL)
  if index
    key = key[0, index].to_sym
  else
    # noop
  end
end

def mm3(method_name)
  key = method_name.to_s
  if key.end_with?(EQUAL)
    key = key.chop.to_sym
  else
    # noop
  end
end

p mm2(:method)
p mm2(:method=)

p mm3(:method)
p mm3(:method=)

Benchmark.ips do |x|
  x.report "mm2 getter" do
    mm2(:method)
  end

  x.report "mm2 setter" do
    mm2(:method=)
  end

  x.report "mm3 getter" do
    mm3(:method)
  end

  x.report "mm3 setter" do
    mm3(:method=)
  end

  x.compare!
end
```

## Results

```
nil
:method
nil
:method
Warming up --------------------------------------
          mm2 getter   213.025k i/100ms
          mm2 setter   151.135k i/100ms
          mm3 getter   227.005k i/100ms
          mm3 setter   169.490k i/100ms
Calculating -------------------------------------
          mm2 getter      3.696M (± 7.0%) i/s -     18.533M in   5.045861s
          mm2 setter      2.146M (± 6.7%) i/s -     10.731M in   5.027797s
          mm3 getter      4.120M (± 8.6%) i/s -     20.430M in   5.010009s
          mm3 setter      2.401M (±11.0%) i/s -     11.864M in   5.021200s

Comparison:
          mm3 getter:  4119677.6 i/s
          mm2 getter:  3695816.6 i/s - same-ish: difference falls within error
          mm3 setter:  2401452.5 i/s - 1.72x  slower
          mm2 setter:  2145832.0 i/s - 1.92x  slower
```